conans/client/conf/detect.py

```diff
@@ -147,7 +147,16 @@ def _get_default_compiler(output):
         output.info("CC and CXX: %s, %s " % (cc or "None", cxx or "None"))
         command = cc or cxx
         if "gcc" in command:
-            return _gcc_compiler(output, command)
+            if platform.system() != "Darwin":
+                return _gcc_compiler(output, command)
+
+            # In Mac OS X check if gcc is a fronted using apple-clang
+            _, out = _execute("%s --version" % command)
+            out = out.lower()
+            if "clang" not in out:
+                return _gcc_compiler(output, command)
+            output.warn("%s detected as a frontend using apple-clang, skipping it to use native apple-clang" % command)
+            command = "clang"  # Force use of clang when gcc is detected as a fronted of apple-clang
```

Comment:
I am not sure we want to fallback to clang. Because if the env-var CC and CXX are set to gcc, then, typically CMake will use them when running, and the user might not notice, because it detected "clang" when CC=gcc. What do you think @lasote?

Reply:
I'm ok with this. If the user is messing with CC/CXX is his problem. And the default profile is just a default.

Reply:
Ok, makes sense